.github/workflows/publish.yaml

```diff
@@ -0,0 +1,86 @@
+name: Publish
+on:
+  push:
+    branches:
+      - 'main'
+      - 'v.*'
+    tags:
+      - '.*'
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.17.1"
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install python3-pip arping ndisc6
+          sudo pip3 install invoke semver pyyaml
+          go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0
+
+      - name: Unit Tests
```

Comment:
Can we leverage the same set of tests we have here https://github.com/metallb/metallb/blob/0cf5b8209c2eb4128f7e64a28615d6a073d10065/.github/workflows/ci.yaml#L47 ?

Reply:
Federico, as we discussed offline, raised an [issue](https://github.com/metallb/metallb/issues/1411) to address this concern as rework has to be done on the CI workflow as well.